### PR TITLE
Add default retries for CI

### PR DESCRIPTION
**Pull request by gardavi**

### CODE REVIEW

File: .gitlab/.gitlab-ci.yml [hunk 1]
```diff
@@ -174,7 +194,7 @@ build featurebase:
 build amd container fb:
   stage: test
   tags:
-    - shell
+    - gdavis-test-shell
```
Comment (JasonAllison) [inline .gitlab/.gitlab-ci.yml:197]: Ia this the name we want to use?

**Review by JasonAllison — COMMENTED** [review]
My only concern is about naming convention

File: .gitlab/.gitlab-ci.yml (hunk 1 above)
Comment (gardavi) [reply .gitlab/.gitlab-ci.yml:197]: That was done just to force running this on a one-off gitlab-runner that I made. https://github.com/molecula/molecular-terraform/pull/307 needs to be merged after this.

**Review by seebs — APPROVED** [review]
Approved if you're doing what I think you're doing. If this would actually cause us to have tests successfully get run, fail, and just rerun them, I am not actually approving that.

File: .gitlab/.gitlab-ci.yml [hunk 2]
```diff
@@ -12,6 +12,16 @@ include:
...
+default:
+  retry:
+    max: 2 # This is confusing but this means "3 runs at max".
+    when:
+      - unknown_failure
+      - api_failure
+      - runner_system_failure
+      - job_execution_timeout
+      - stuck_or_timeout_failure
+
```
[the remark this answers is not in the document]
Comment (gardavi) [reply .gitlab/.gitlab-ci.yml:24]: Yeah this is misc failures that are not the tests failing. The biggest one that occurs right now is `runner_system_failure` which happens if one of the spot instances gets terminated while running a job.

[the remark this answers is not in the document]
Comment (gardavi) [reply .gitlab/.gitlab-ci.yml:24]: The reference for these is here: https://docs.gitlab.com/ee/ci/yaml/#retrywhen

[the remark this answers is not in the document]
Comment (gardavi) [reply .gitlab/.gitlab-ci.yml:24]: The only one I'm a bit unsure about is `job_execution_timeout`: Retry if the script exceeded the maximum execution time set for the job.

[the remark this answers is not in the document]
Comment (seebs) [reply .gitlab/.gitlab-ci.yml:24]: i think i'm okay with that one because our test timeouts are shorter than the execution time limit anyway.

**Review by seebs — APPROVED** [review]

**Pull request merged**